src/table/internal.tsx

```diff
@@ -363,26 +364,16 @@ const InternalTable = React.forwardRef(
                 <tbody>
                   {loading || items.length === 0 ? (
                     <tr>
-                      <td
-                        colSpan={totalColumnsCount}
-                        className={clsx(styles['cell-merged'], hasFooter && styles['has-footer'])}
-                      >
-                        <div
-                          className={styles['cell-merged-content']}
-                          style={{
-                            width:
-                              (supportsStickyPosition() && containerWidth && Math.floor(containerWidth)) || undefined,
-                          }}
-                        >
-                          {loading ? (
-                            <InternalStatusIndicator type="loading" className={styles.loading} wrapText={true}>
-                              <LiveRegion visible={true}>{loadingText}</LiveRegion>
-                            </InternalStatusIndicator>
-                          ) : (
-                            <div className={styles.empty}>{empty}</div>
-                          )}
-                        </div>
-                      </td>
+                      <EmptyStateCell
```

Comment:
[minor naming] I don't think it's ideal that `EmptyStateCell` also renders the loading state, perhaps something like `FullTableCell` or `NoDataCell` instead (not sure either is ideal though, open to other suggestions!)

Reply:
I like the `NodeDataCell` a lot - will update 👍